oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -163,6 +167,15 @@ public void refresh() throws IOException {
     }
   }
 
+  /**
+   * Provide additional headers to return as request metadata.
+   *
+   * @return Map of additional headers.
+   */
+  protected Map<String, List<String>> getAdditionalHeaders() {
```

Comment:
nested types (Map of List) suggest there's a missed opportunity for a more semantic class.

Reply:
HTTP headers behave as maps of `String`s to a list of `String`s. We could possible wrap it, but we're not adding any extra behavior. For a general purpose HTTP client, we might do so, but it seem unnecessary here.

Note: `getRequestMetadata()` does return this same type (but we really can't feasibly change that as everything that depends on google-auth-library expects that return type).